### ISSUE TITLE
AWS ECS documentation

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#4105]
I'm opening this issue to represent a single, final, place for AWS ECS documentation. This issue will be used to build out the website, docs, and marketing pages.

- [ ] Include a step-by-step guide for deploying Vector for ECS on EC2
- [x] Include a step-by-step guide for deploying Vector for ECS on Fargate
- [ ] Add sections any other ECS specific details that might be relevant for marketing or documentation. Please organize these into distinct sections.

**Comment by awangc** [comment]
@ktff Is there any networking mode that needs be configured? I tested the setup you described (just used port 8080 instead of 8088), and I am seeing:

```
ResourceInitializationError: failed to validate logger args: Options http://0.0.0.0:8080/services/collector/event/1.0: dial tcp 0.0.0.0:8080: connect: connection refused : exit status 1
```
I am using `timberio/vector:0.10.0-debian` as vector image

**Comment by ktff** [comment]
@awangc did you add 
```
  address = "0.0.0.0:8080" 
```
to configuration of `splunk_hec`. If that address isn't specified then the source will use `8088` port by default. 

Regarding networking mode, on Fargate only supported mode is `awsvpc` so the guide assumes that. EC2 is a slightly different story.

For EC2 with different networking mode you will need to ensure that the port of Vector container is accessible from your containers which would possibly require changing contents of Vector's `portMappings`. So the container would end up with something like this
```json
"portMappings": [
  {
    "hostPort": 8080,
    "protocol": "tcp",
    "containerPort": 8080
  }
]
```

**Comment by awangc** [comment]
@ktff Yes, I have that part in the vector.toml file:

```
[sources.app_log]
  type = "splunk_hec"
  address = "0.0.0.0:8080"
  token = "aabbccddeeff"
```

**Comment by awangc** [comment]
@ktff I found out the setup does not work for Fargate platform 1.4 but works for `LATEST` (1.3 if I'm not mistaken). Possible reason is that container start order is not being respected https://github.com/aws/containers-roadmap/issues/849 ?

**Comment by ktff** [comment]
> Possible reason is that container start order is not being respected

Yes, that explains the error message. The app container logger is trying to establish connection before starting app container, and if the the vector container isn't running at that time, there is nothing listening. It also seams that they are trying to connect only once, at the time of the start. In that case there are two options:
* Find a way to enforce ordering.
* Deploy vector somewhere else before starting task and ensure that it's reachable.

**Comment by leobudima** [comment]
> @ktff I found out the setup does not work for Fargate platform 1.4 (which is the one I had been trying) but works for `LATEST` (1.3 if I'm not mistaken). Possible reason is that container start order is not being respected [aws/containers-roadmap#849](https://github.com/aws/containers-roadmap/issues/849) ?

Thank you for your insight, I have the same issue - did you manage to work around this on 1.4?

**Comment by gtorre** [comment]
Any updates here? Thanks!

**Comment by thedannywilcox** [comment]
Found this coming from the internet while working on getting it setup. Not sure if everything works, but I was able to get Vector setup on the LATEST Fargate (1.4) without issue using File source and cloudwatch sink.

**Comment by edeno1** [comment]
I'm trying to make it the container stops working

**Comment by madan-wego** [comment]
By Adding splunk-verify-connection": "false" it worked for me in AWS Fargate and Ec2.

logConfiguration": {
        "logDriver": "splunk",
        "options": {
          "splunk-url": "http://0.0.0.0:8088",
          "splunk-verify-connection": "false",
          "splunk-token": "abc1234567890"
        }
      }
